tests/scipy_stats_test.py

```diff
@@ -472,6 +472,143 @@ def args_maker():
       self._CheckAgainstNumpy(scipy_fun, lax_fun, args_maker, tol=1e-4)
       self._CompileAndCheck(lax_fun, args_maker, rtol=3e-4)
 
+  @genNamedParametersNArgs(5)
+  def testTruncnormLogPdf(self, shapes, dtypes):
+    rng = jtu.rand_default(self.rng())
+    scipy_fun = osp_stats.truncnorm.logpdf
+    lax_fun = lsp_stats.truncnorm.logpdf
+
+    def args_maker():
+      x, a, b, loc, scale = map(rng, shapes, dtypes)
+
+      # hack to ensure that b > a and a < x < b
+      a = -np.abs(a).astype(a.dtype)
+      x = ((a + b) / 2).astype(x.dtype)
+      b = np.abs(b).astype(b.dtype)
```

Comment:
What happens if these are not in order `a < x < b`? It should return `-inf`, correct? Should we just let the values be out of order so as to test that?

If we need them in order, it might be simpler to do something like this, and then cast to the desired dtype:
```python
a, x, b = rng((3, *shape), dtype).sort(0)
```

Reply:
Yea, true, I guess we can relax the need for `x` to be in bounds, but `b` must be greater than `a`. But since `a` or `b` can have different shapes, I couldn't figure out a clean way to use `np.sort()` to get this to work. My hack might be the simplest way of enforcing this, but I could think more about it!

Reply:
It looks like when `a >= b` the scipy version returns `nan`:
```python
from scipy.stats import truncnorm
print(truncnorm.logpdf(1, a=2, b=0))
# nan
```
I believe `_CheckAgainstNumpy` will check the locations of NaN values, so it might be best in the tests to check the full range of values to ensure that we're matching that behavior.